resources/js/features/game-list/components/WantToPlayGamesDataTable/WantToPlayGamesDataTable.tsx

```diff
@@ -0,0 +1,162 @@
+import { useMemo } from 'react';
+
+import {
+  BaseTable,
+  BaseTableBody,
+  BaseTableCell,
+  BaseTableHead,
+  BaseTableHeader,
+  BaseTableRow,
+} from '@/common/components/+vendor/BaseTable';
+import { useGameListQuery } from '@/common/hooks/useGameListQuery';
+import { usePageProps } from '@/common/hooks/usePageProps';
+import { cn } from '@/utils/cn';
+
+import { buildColumnDefinitions } from './buildColumnDefinitions';
+import { DataTablePagination } from './DataTablePagination';
+import { WantToPlayGamesDataTableToolbar } from './WantToPlayGamesDataTableToolbar';
+
+// These values are all generated from `useGameListState`.
+interface WantToPlayGamesDataTableProps {
+  columnFilters: ColumnFiltersState;
+  columnVisibility: VisibilityState;
+  pagination: PaginationState;
+  setColumnFilters: Dispatch<SetStateAction<ColumnFiltersState>>;
+  setColumnVisibility: Dispatch<SetStateAction<VisibilityState>>;
+  setPagination: Dispatch<SetStateAction<PaginationState>>;
+  setSorting: Dispatch<SetStateAction<SortingState>>;
+  sorting: SortingState;
+}
+
+export const WantToPlayGamesDataTable: FC<WantToPlayGamesDataTableProps> = ({
+  columnFilters,
+  columnVisibility,
+  pagination,
+  setColumnFilters,
+  setColumnVisibility,
+  setPagination,
+  setSorting,
+  sorting,
+}) => {
+  const { can } = usePageProps<App.Community.Data.UserGameListPageProps>();
+
+  const gameListQuery = useGameListQuery({ columnFilters, pagination, sorting });
+
+  const table = useReactTable({
+    columns: useMemo(
+      () =>
+        buildColumnDefinitions({
+          canSeeOpenTicketsColumn: can.develop ?? false,
+        }),
+      [can.develop],
+    ),
+    data: gameListQuery.data?.items ?? [],
+    manualPagination: true,
+    manualSorting: true,
+    manualFiltering: true,
+    rowCount: gameListQuery.data?.total,
+    pageCount: gameListQuery.data?.lastPage,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnFiltersChange: (updateOrValue) => {
+      table.setPageIndex(0);
+
+      setColumnFilters(updateOrValue);
+    },
+    onPaginationChange: (newPaginationValue) => {
+      setPagination(newPaginationValue);
+    },
+    onSortingChange: (newSortingValue) => {
+      table.setPageIndex(0);
+
+      setSorting(newSortingValue);
+    },
+    getCoreRowModel: getCoreRowModel(),
+    state: { columnFilters, columnVisibility, pagination, sorting },
+  });
+
+  const visibleColumnCount = table.getVisibleFlatColumns().length;
+
+  return (
+    <div className="flex flex-col gap-3">
+      <WantToPlayGamesDataTableToolbar
+        table={table}
+        unfilteredTotal={gameListQuery.data?.unfilteredTotal ?? null}
+      />
+
+      <BaseTable
+        containerClassName={cn(
+          'overflow-auto rounded-md border border-neutral-700/80 bg-embed',
+          'light:border-neutral-300 lg:overflow-visible lg:rounded-sm',
+
+          // A sticky header cannot support this many columns. We have to drop stickiness.
```

Comment:
It's worth nothing that sticky headers have to be disabled due to CSS restrictions once the user enables a certain column count on either LG or XL.